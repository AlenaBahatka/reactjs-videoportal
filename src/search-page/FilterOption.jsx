import React from 'react';

function FilterOption(props) {
    return (
        <label className="pink">
        <input 
            type="radio" 
            name={props.optionName} 
            checked={props.checkedOption === props.name}
            value={props.name}
            onChange={(event) => props.handleOptionChange(event.target.value)}
        />
        <span>{props.displayName.toUpperCase()}</span>
    </label>
    );
}

export default FilterOption;
