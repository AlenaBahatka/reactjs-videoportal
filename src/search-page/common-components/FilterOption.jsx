// @flow
import React from 'react';

type FilterOptionPropTypes = {
	optionName: string,
	checkedOption: string,
	name: string,
	displayName: string,
	handleOptionChange: Function
}

function FilterOption(props: FilterOptionPropTypes) {
	return (
		<label>
			<input
				type="radio"
				name={props.optionName}
				checked={props.checkedOption === props.name}
				value={props.name}
				onChange={(event) => props.handleOptionChange(event.target.value)}
			/>
			<span>{props.displayName.toUpperCase()}</span>
		</label>
	);
}

export default FilterOption;
