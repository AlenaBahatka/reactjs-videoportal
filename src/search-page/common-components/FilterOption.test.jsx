import React from 'react';
import { shallow, mount } from 'enzyme';
import FilterOption from './FilterOption';

let handleOptionChangeMock, filmPageComponent, filterOption;

describe('FilterOption', () => {
	beforeAll(() => {
		handleOptionChangeMock = jest.fn();
		filterOption = (
			<FilterOption
				optionName="title"
				checkedOption="title"
				displayName="title"
				name="title"
				handleOptionChange={handleOptionChangeMock}
			/>
		);
	});

	it('should correctly render component', () => {
		filmPageComponent = shallow(filterOption);
		expect(filmPageComponent).toMatchSnapshot();
	});

	it('should call method on event change', () => {
		filmPageComponent = mount(filterOption);

		filmPageComponent.find('input').simulate('change');

		expect(handleOptionChangeMock).toBeCalled();
	});
});
