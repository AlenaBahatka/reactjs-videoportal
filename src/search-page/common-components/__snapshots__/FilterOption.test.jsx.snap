// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterOption should correctly render component 1`] = `
<label>
  <input
    checked={true}
    name="title"
    onChange={[Function]}
    type="radio"
    value="title"
  />
  <span>
    TITLE
  </span>
</label>
`;
